src/transports/ws/sws.c

```diff
@@ -773,14 +744,21 @@ static int nn_sws_fail_conn (struct nn_sws *self, int code, char *reason)
 
     self->fail_msg_len += payload_len;
 
-    self->instate = NN_SWS_INSTATE_FAILING;
+    self->instate = NN_SWS_INSTATE_CLOSED;
 
-    /*  On the next recv, the connection will be failed. Why defer
-        until the next recv? Semantically, until then, this incoming
-        message has not been interpreted, so it's not until then that
-        it could be failed. This type of pre-processing is necessary
-        to early fail chunked transfers. */
```

Comment:
What you're saying does not sound wrong ...

... but what I can say is that at the time, when developing this part that feels/looks wonky compared to your edit, it was done for a reason.

Now, I cannot say if that reason was a good reason, or even if it was then, is now *still* a good reason -- but it was definitely to address some clever corner-case that the Autobahn fuzzer had concocted.

Or maybe, it was to address a deficiency in the sync/async send/recv capabilities of libnanomsg vs requirements here ... I can't quite recall ... this was almost a year ago, and I'll do everything I can to mentally spool up again to match your renewed vigor here. :+1:

(For context about the talk about the "Autobahn" tester, and also apparent "lack" of websocket testing in libnanomsg -- there exists a PR I created that provides a comprehensive RFC 6455 fuzzer and stress tester, except we debated whether this PR would be necessary for libnanomsg or not. The arguments against where 1) it introduces new dependencies, such as Python and Python packages, and 2) libnanomsg does not attempt to be, and specifically should not attempt to be, fully RFC 6455 compliant. I agree with these two arguments.)